test/Constraints/identical-type-name.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift-dylib(%t/%target-library-name(A)) -module-name A -emit-module -emit-module-path %t/A.swiftmodule %S/Inputs/identical-type-name-module.swift
+// RUN: %target-build-swift-dylib(%t/%target-library-name(B)) -module-name B -emit-module -emit-module-path %t/B.swiftmodule %S/Inputs/identical-type-name-module.swift
+// RUN: %target-typecheck-verify-swift -I %t
+
+import A
+import B
+
+func f<T>(_ type: T.Type) {}
+
+func test() {
+  f(SomeStruct.self) // expected-error {{conflicting arguments to generic parameter 'T' ('A.SomeStruct' vs. 'B.SomeStruct')}}
```

Comment:
Can you try a few more examples? Ideas: failing conversions from Foo to Foo, mismatched concrete parameter types, etc?

Reply:
Good idea! I added many more tests and applied the same fix to `diagnoseContextualFunctionCallGenericAmbiguity`.